### PR TITLE
refactor-codebase

### LINKED ISSUE #46: Issue: Validation to member address
Add validation to member address in the frontend and the backend.

### PR DESCRIPTION
**Pull request by dasarathg68**
# Description
Adds address validation to both frontend and backend and performs refactoring of the codebase

Fixes #46 
#40 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

* **Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines

### CODE REVIEW

**Review by farrellh1 — APPROVED** [review]
LGTM, wait for herman approval

File: app/src/utils/walletValidatorUtil.ts
Comment (hermannleboss) [inline app/src/utils/walletValidatorUtil.ts:1]: No need to create it

**Review by hermannleboss — CHANGES_REQUESTED** [review]
Use ether.utils and emit event instead of using it in the component

**Review by dasarathg68 — COMMENTED** [review]
Done with ethers.utils and refactoring

File: app/src/components/MemberCard.vue
Comment (farrellh1) [inline app/src/components/MemberCard.vue:12]: we can change this modal dialog to be component

File: app/src/components/MemberCard.vue
Comment (farrellh1) [inline app/src/components/MemberCard.vue:35]: and also we can refactor this button into single component, and the props will be the color of the button and the size and the function of the click

File: app/src/types/types.ts
Comment (farrellh1) [inline app/src/types/types.ts:20]: i think we can seperate each type into single file, just my opinion

Comment (farrellh1) [reply app/src/types/types.ts:20]: and then add into `index.ts` to do barrel export https://medium.com/@klauskpm/do-a-barrel-export-aa5b79b76b05

**Review by farrellh1 — COMMENTED** [review]
You should also remove the `console.log` because we shouldnt show any log in production

File: app/src/components/MemberCard.vue
Comment (dasarathg68) [reply app/src/components/MemberCard.vue]: I think daisy ui already simplifies this for us and the whole point of Tailwind and Daisy UI is to define class names on the go.
A generic component, I feel, would take away the point of it.

File: backend/src/controllers/memberController.ts
Comment (aliassan) [inline backend/src/controllers/memberController.ts:38]: `errorResponse` util takes care of this for you including conditionally printing in dev mode. We standardised to avoid leaking sensitive information about the back-end architecture to the front-end

File: app/src/apis/memberApi.ts
Comment (aliassan) [inline app/src/apis/memberApi.ts]: I feel you're violating Single Responsibility here, you're making API calls, displaying errors to the UI and handling errors at the same. You should just throw an error and have some lower level code somewhere decide what to do with the error. The way you're doing it forces you to go and implement this logic in all your API code instead of just where the API classes are used.

Comment (dasarathg68) [reply app/src/apis/memberApi.ts]: The show function is from useToastStore (not a Vue function) and it keeps UI interactions separate from component logic, and although it might seem like error handling occurs within the component, show is still a lower-level code like you mentioned, fetched from the centralized Pinia store.

Comment (aliassan) [reply app/src/apis/memberApi.ts]: Yes, you have a `try..catch` block around your code which is error handling. Instead you would have that where you are "using" `FetchMemberAPI` which is most likely going to be a composable soon although, right now you're calling it directly in a view.

File: app/src/apis/memberApi.ts
Comment (aliassan) [inline app/src/apis/memberApi.ts:27]: I suggest that you just throw the error and handle them where you're using or calling the classes, it's simpler.

**Review by hermannleboss — APPROVED** [review]

**Pull request merged**
